### PR TITLE
[3.3] [Deprecations] Switch YamlUpdater to use Bolt filesystem for YAML handling

### PR DESCRIPTION
**Pull request by GawainLynch**
Cuts out deprecations in SF 3.1+ too

### CODE REVIEW

**Comment by GawainLynch** [comment]
Side note: 
```bash
$ ./app/nut config:get thumbnails/aliases/myimageformat/cropping -f theme://theme.yml
thumbnails/aliases/myimageformat/cropping: crop
```

File: src/Configuration/YamlUpdater.php [hunk 1]
```diff
@@ -15,34 +14,28 @@
...
-        $this->parser = new Parser();
-
-        // Get the contents of the file
-        $yaml = $this->file->read();
-
-        // Check that the read-in YAML is valid
-        $this->parsed = $this->parser->parse($yaml, true, true);
+        if (is_string($file)) {
+            $fs = $app['filesystem']->getFilesystem('config');
+            $file = $fs->getFile($file);
```
Comment (CarsonF) [inline src/Configuration/YamlUpdater.php]: Why not:
```php
$file = $app['filesystem']->getFile('config://' . $file);
```

File: src/Configuration/YamlUpdater.php [hunk 2]
```diff
@@ -156,7 +149,7 @@ protected function verify()
         }
 
         // This will throw a ParseException If the YAML is not valid
-        $this->parser->parse($this->yaml, true, true);
+        $this->parsed = $this->file->parse(['exceptionsOnInvalidType' => true, 'objectSupport' => true]);
```
Comment (CarsonF) [inline src/Configuration/YamlUpdater.php]: This won't work as the file doesn't have the new contents yet.

File: src/Configuration/YamlUpdater.php (hunk 1 above)
Comment (GawainLynch) [inline src/Configuration/YamlUpdater.php]: Throw … exception

File: src/Configuration/YamlUpdater.php (hunk 2 above)
Comment (GawainLynch) [inline src/Configuration/YamlUpdater.php]: Ah!

File: src/Configuration/YamlUpdater.php (hunk 1 above)
Comment (CarsonF) [inline src/Configuration/YamlUpdater.php]: No it doesn't.

Comment (GawainLynch) [inline src/Configuration/YamlUpdater.php]: Oh, nvm

File: src/Configuration/YamlUpdater.php (hunk 2 above)
Comment (GawainLynch) [inline src/Configuration/YamlUpdater.php]: Done

File: src/Configuration/YamlUpdater.php (hunk 1 above)
Comment (GawainLynch) [inline src/Configuration/YamlUpdater.php]: Done

**Comment by GawainLynch** [comment]
> Can ConfigGet/Set be abstracted? The majority of the lines in both of those files are the same.

Yeah, I did wonder … I might in the morning

**Comment by GawainLynch** [comment]
>> Can ConfigGet/Set be abstracted? The majority of the lines in both of those files are the same.

> Yeah, I did wonder … I might in the morning

Done

**Comment by bobdenotter** [comment]
This should be GTG now, right?

**Comment by GawainLynch** [comment]
> This should be GTG now, right?

/me nudges @CarsonF

**Pull request merged**